File: client/app/src/commonMain/kotlin/com/jahirtrap/cconnect/ui/ComponentIcons.kt
package com.jahirtrap.cconnect.ui

import androidx.compose.ui.graphics.vector.ImageVector
import com.composables.icons.lucide.Check
import com.composables.icons.lucide.CircleQuestionMark
import com.composables.icons.lucide.Clock
import com.composables.icons.lucide.Download
import com.composables.icons.lucide.ExternalLink
import com.composables.icons.lucide.File
import com.composables.icons.lucide.Folder
import com.composables.icons.lucide.Info
import com.composables.icons.lucide.Lightbulb
import com.composables.icons.lucide.Lucide
import com.composables.icons.lucide.MessageSquare
import com.composables.icons.lucide.Pencil
import com.composables.icons.lucide.Plus
import com.composables.icons.lucide.RefreshCw
import com.composables.icons.lucide.Search
import com.composables.icons.lucide.Settings
import com.composables.icons.lucide.Shield
import com.composables.icons.lucide.Sparkles
import com.composables.icons.lucide.Trash2
import com.composables.icons.lucide.TriangleAlert
import com.composables.icons.lucide.X

fun componentIcon(name: String?): ImageVector? = when (name) {
    "question" -> Lucide.CircleQuestionMark
    "clock" -> Lucide.Clock
    "sparkles" -> Lucide.Sparkles
    "message-square" -> Lucide.MessageSquare
    "check" -> Lucide.Check
    "x" -> Lucide.X
    "plus" -> Lucide.Plus
    "pencil" -> Lucide.Pencil
    "trash" -> Lucide.Trash2
    "download" -> Lucide.Download
    "external-link" -> Lucide.ExternalLink
    "refresh" -> Lucide.RefreshCw
    "search" -> Lucide.Search
    "settings" -> Lucide.Settings
    "info" -> Lucide.Info
    "alert" -> Lucide.TriangleAlert
    "lightbulb" -> Lucide.Lightbulb
    "shield" -> Lucide.Shield
    "file" -> Lucide.File
    "folder" -> Lucide.Folder
    else -> null
}

File: tauri/src/lib/ui/componentIcons.ts
import Check from "@lucide/svelte/icons/check";
import CircleQuestionMark from "@lucide/svelte/icons/circle-question-mark";
import Clock from "@lucide/svelte/icons/clock";
import Sparkles from "@lucide/svelte/icons/sparkles";
import Download from "@lucide/svelte/icons/download";
import ExternalLink from "@lucide/svelte/icons/external-link";
import File from "@lucide/svelte/icons/file";
import Folder from "@lucide/svelte/icons/folder";
import Info from "@lucide/svelte/icons/info";
import Lightbulb from "@lucide/svelte/icons/lightbulb";
import MessageSquare from "@lucide/svelte/icons/message-square";
import Pencil from "@lucide/svelte/icons/pencil";
import Plus from "@lucide/svelte/icons/plus";
import RefreshCw from "@lucide/svelte/icons/refresh-cw";
import Search from "@lucide/svelte/icons/search";
import Settings from "@lucide/svelte/icons/settings";
import Shield from "@lucide/svelte/icons/shield";
import Trash2 from "@lucide/svelte/icons/trash-2";
import TriangleAlert from "@lucide/svelte/icons/triangle-alert";
import X from "@lucide/svelte/icons/x";
import type { IconSource } from "./icons";

const ICONS: Record<string, IconSource> = {
  question: CircleQuestionMark,
  clock: Clock,
  sparkles: Sparkles,
  "message-square": MessageSquare,
  check: Check,
  x: X,
  plus: Plus,
  pencil: Pencil,
  trash: Trash2,
  download: Download,
  "external-link": ExternalLink,
  refresh: RefreshCw,
  search: Search,
  settings: Settings,
  info: Info,
  alert: TriangleAlert,
  lightbulb: Lightbulb,
  shield: Shield,
  file: File,
  folder: Folder,
};

export const componentIcon = (name: string | null): IconSource | null => (name ? (ICONS[name] ?? null) : null);
